test/testsupport/stubs/WireMockHelpers.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testsupport.stubs
+
+import cats.syntax.eq._
+import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, equalTo, post, postRequestedFor, stubFor, urlPathEqualTo, verify}
+import com.github.tomakehurst.wiremock.http.Request
+import com.github.tomakehurst.wiremock.matching.MatchResult
+import com.github.tomakehurst.wiremock.stubbing.StubMapping
+import play.api.http.Status
+import play.api.libs.json.{Format, Json}
+
+object WireMockHelpers {
+
+  /**
+   * Useful wiremock helper to verify that the request body serialises to what we expect.
+   * Hint: If it's not working and you can't work out why, check what A you are passing in... :)
+   * @param url: String
+   * @param format: play.api.libs.json.Format as an implicit
+   * @tparam A: Model we want to get format for
+   */
+  def verifyWithBodyParse[A](url: String)(implicit format: Format[A]): Unit = verify(
+    postRequestedFor(urlPathEqualTo(url))
+      .andMatching((value: Request) => customValueMatcher(url, value))
+  )
+
+  /**
+   * Same as above, but overloaded with headers to check
+   */
+  def verifyWithBodyParse[A](url: String, headers: (String, String))(implicit format: Format[A]): Unit = verify(
+    postRequestedFor(urlPathEqualTo(url))
+      .withHeader(headers._1, equalTo(headers._2))
+      .andMatching((value: Request) => customValueMatcher(url, value))
+  )
+
+  /**
+   * Helper method to attempt to parse request body into type, using asOpt.
+   * If option is empty, it can't and wiremock verify should fail, else wiremock verify is successful.
+   */
+  private def customValueMatcher[A](url: String, request: Request)(implicit format: Format[A]): MatchResult =
```

Comment:
there's an extra layer of validation we could do which is to pass in the expected `A` and compare it with what was actually found in the request - what do you think?

Reply:
That's true, but I was trying to avoid having to add tons of extra json etc - I guess I can add this though